gazebo_plugins/src/gazebo_ros_depth_camera.cpp

```diff
@@ -381,14 +397,73 @@ bool GazeboRosDepthCamera::FillPointCloudHelper(
     uint32_t rows_arg, uint32_t cols_arg,
     uint32_t step_arg, void* data_arg)
 {
+  // Skip if color image has not been filled yet
+  if (this->image_msg_.data.empty())
+  {
+    ROS_DEBUG_STREAM_NAMED("depth_camera", "Image is empty, skipping point cloud generation. "
+                           << "This should only happen once at the beginning");
+    return false;
```

Comment:
The very first time false is returned, this it not checked by FillPointdCloud.
```

  ///copy from depth to point cloud message
   FillPointCloudHelper(this->point_cloud_msg_,
                  this->height,
                  this->width,
                  this->skip_,
                  (void*)_src );
 
    this->point_cloud_pub_.publish(this->point_cloud_msg_);
```
This results in publishing an empty point cloud with no fields in the beginning. Maybe change to:
```
///copy from depth to point cloud message
   if(FillPointCloudHelper(this->point_cloud_msg_,
                  this->height,
                  this->width,
                  this->skip_,
                  (void*)_src )){
       this->point_cloud_pub_.publish(this->point_cloud_msg_);
   }
```